qiita_db/test/test_analysis.py

```diff
@@ -117,15 +123,98 @@ def test_retrieve_samples(self):
+            "number_samples_promised": 28,
+            "portal_type_id": 3,
+            "study_alias": "FCM",
+            "study_description": "Microbiome of people who eat nothing but "
+                                 "fried chicken",
+            "study_abstract": "Exploring how a high fat diet changes the "
+                              "gut microbiome",
+            "emp_person_id": StudyPerson(2),
+            "principal_investigator_id": StudyPerson(3),
+            "lab_person_id": StudyPerson(1)
+        }
+        metadata_dict = {
+            'SKB8.640193': {'physical_location': 'location1',
+                            'has_physical_specimen': True,
+                            'has_extracted_data': True,
+                            'sample_type': 'type1',
+                            'required_sample_info_status': 'received',
+                            'collection_timestamp':
+                            datetime(2014, 5, 29, 12, 24, 51),
+                            'host_subject_id': 'NotIdentified',
+                            'Description': 'Test Sample 1',
+                            'str_column': 'Value for sample 1',
+                            'latitude': 42.42,
+                            'longitude': 41.41},
+            'SKD8.640184': {'physical_location': 'location1',
+                            'has_physical_specimen': True,
+                            'has_extracted_data': True,
+                            'sample_type': 'type1',
+                            'required_sample_info_status': 'received',
+                            'collection_timestamp':
+                            datetime(2014, 5, 29, 12, 24, 51),
+                            'host_subject_id': 'NotIdentified',
+                            'Description': 'Test Sample 2',
+                            'str_column': 'Value for sample 2',
+                            'latitude': 4.2,
+                            'longitude': 1.1},
+            'SKB7.640196': {'physical_location': 'location1',
+                            'has_physical_specimen': True,
+                            'has_extracted_data': True,
+                            'sample_type': 'type1',
+                            'required_sample_info_status': 'received',
+                            'collection_timestamp':
+                            datetime(2014, 5, 29, 12, 24, 51),
+                            'host_subject_id': 'NotIdentified',
+                            'Description': 'Test Sample 3',
+                            'str_column': 'Value for sample 3',
+                            'latitude': 4.8,
+                            'longitude': 4.41},
+            }
+        metadata = pd.DataFrame.from_dict(metadata_dict, orient='index')
+
+        Study.create(User("test@foo.bar"), "Test study 2", [1], info)
+
+        SampleTemplate.create(metadata, Study(2))
+
+        mp = get_mountpoint("processed_data")[0][1]
         try:
```

Comment:
After going over the code with @squirrelo we agreed is fine but I think we need a small block of text, comment, to explain why we have these try/finally so when a new dev needs to change this code it reads an explanation vs. needing to figure it out by some intraline one liner comments. Note that this test only became clear to me after an online explanation and I think is worth to have it documented so this doesn't get lost.